temporalio/workflow.py

```diff
@@ -746,6 +771,121 @@ def time_ns() -> int:
     return _Runtime.current().workflow_time_ns()
 
 
+# Needs to be defined here to avoid a circular import
+@runtime_checkable
+class UpdateMethodMultiArg(Protocol[MultiParamSpec, ProtocolReturnType]):
+    """Decorated workflow update functions implement this."""
+
+    _defn: temporalio.workflow._UpdateDefinition
+
+    def __call__(
+        self, *args: MultiParamSpec.args, **kwargs: MultiParamSpec.kwargs
+    ) -> Union[ProtocolReturnType, Awaitable[ProtocolReturnType]]:
+        """Generic callable type callback."""
+        ...
+
+    def validator(self, vfunc: Callable[MultiParamSpec, None]) -> None:
```

Comment:
Hrmm, usually the decorator has to return the function back (https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators). Probably won't matter for most since they never reference it, but if users wanted to also use their own decorator with the validator decorator they'd expect the validator decorator to return the function back.

Reply:
Sure, can return it, no reason not to I suppose